### PR TITLE
fix: don't set http.redirect_codes if the attr doesn't exist

### PR DESCRIPTION
**Pull request by busunkim96**
This allows folks who can't upgrade their httplib2 version to use the latest version of the library. Note that it is still strongly recommended that you upgrade (see [install_requires](https://github.com/googleapis/google-api-python-client/blob/master/setup.py#L36)). 

@aaltay is there an issue tracking the httplib2 upgrade I can link to? I'd like this to be removed eventually.

### CODE REVIEW

**Comment by aaltay** [comment]
Could we also change install requires to be similar to before like: `"httplib2>=0.9.2,<1dev"`. Because, now with this change the library will be backward compatible, and it will not cause force upgrades of httplib2 library. (force upgrades also cause issues.) ?

**Comment by busunkim96** [comment]
Made the change to setup.py.

Is it possible for your tests to pin to an older version of this library until you're able to sort out upgrading httplib2? Or is there a requirement for you to always be testing with the latest version of google-api-python-client?

**Comment by aaltay** [comment]
> Made the change to setup.py.
> 
> Is it possible for your tests to pin to an older version of this library until you're able to sort out upgrading httplib2? Or is there a requirement for you to always be testing with the latest version of google-api-python-client?

I don't there is a requirement to use at head. This is our setup file (https://github.com/apache/beam/blob/master/sdks/python/setup.py) -- I cannot find a direct dependency. Probably there is some indirect dependency.

File: setup.py [hunk 1]
```diff
@@ -33,7 +33,7 @@
 packages = ["apiclient", "googleapiclient", "googleapiclient/discovery_cache"]
 
 install_requires = [
-    "httplib2>=0.17.0,<1dev",
+    "httplib2>=0.9.2,<1dev",
```
Comment (crwilcox) [inline setup.py]: should we add a comment on this line so folks don't mistakenly try to update it?

Comment (busunkim96) [reply setup.py]: Done.

File: setup.py [hunk 2]
```diff
@@ -33,7 +33,10 @@
 packages = ["apiclient", "googleapiclient", "googleapiclient/discovery_cache"]
 
 install_requires = [
-    "httplib2>=0.17.0,<1dev",
+    # NOTE: Apache Beam tests depend on this library and cannot currently
+    # currently upgrade their httplib2 version.
```
Comment (daikeshi) [inline setup.py]: nit: two "currently" here.

**Comment by aaltay** [comment]
Thank you all! When would this change be available in pypi?

**Comment by busunkim96** [comment]
Just looking for an approval on the release PR(#842) and then I'll push it out to PyPI. :)

**Comment by busunkim96** [comment]
1.8.0 is released @aaltay https://pypi.org/project/google-api-python-client/1.8.0/

**Comment by aaltay** [comment]
> 1.8.0 is released @aaltay https://pypi.org/project/google-api-python-client/1.8.0/

Thank you very much for the quick response!

**Pull request merged**